ci: add documentation regarding secrets management (#25607)
sourcegraph/sourcegraph: enterprise/dev/ci/README.md

go run ./enterprise/dev/ci/gen-pipeline.go | buildkite-agent pipeline upload
```

> ## Secrets managements

Reviewer: ```suggestion
## Managing secrets
```

> Our CI pipeline must never leak secrets:

Reviewer: ```suggestion
The term *secret* refers to authentication credentials like passwords, API keys, tokens, etc. which are used to access a particular service. Our CI pipeline must never leak secrets:
```

> - to add a secret, use the Secret Manager on Google Cloud and then inject it at deployment time as an environment variable in the CI agents, which will make it available to every steps.

Reviewer: ```suggestion
- to add a secret, use the Secret Manager on Google Cloud and then inject it at deployment time as an environment variable in the CI agents, which will make it available to every step.
```

> - use an environment variable name suffixed as following to ensure it gets redacted in the logs: `*_PASSWORD,*_SECRET,*_TOKEN,*_ACCESS_KEY,*_SECRET_KEY,*_CREDENTIALS`

Reviewer: ```suggestion
- use an environment variable name with one of the following suffixes to ensure it gets redacted in the logs: `*_PASSWORD, *_SECRET, *_TOKEN, *_ACCESS_KEY, *_SECRET_KEY, *_CREDENTIALS`
```

> - while environment variables can be assigned when declaring steps, they should never be used for secrets:

Reviewer: ```suggestion
- while environment variables can be assigned when declaring steps, they should never be used for secrets, because they won't get redacted, even if they match one of the above patterns.
```

>   - they won't get redacted, even if they match the above pattern.

Reviewer: ```suggestion
```
